### PR TITLE
Fix wallet locking behavior

### PR DESCRIPTION
**Pull request by bvbfan**
- Fix build with newer boost

- Wallet should be unlocked only in when it sign a transaction

- Small refactor as well as small .yaml syntax correction

### CODE REVIEW

File: src/httpserver.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <sync.h>
 #include <ui_interface.h>
 
+#include <deque>
```
Comment (ShengguangXiao) [inline src/httpserver.cpp:18]: Why need to add this include?

File: src/masternodes/mn_rpc.cpp [hunk 2]
```diff
@@ -310,7 +312,7 @@ UniValue resignmasternode(const JSONRPCRequest& request) {
             throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("The masternode %s does not exist", nodeIdStr));
         }
         ownerDest = nodePtr->ownerType == 1 ? CTxDestination(PKHash(nodePtr->ownerAuthAddress)) : CTxDestination(WitnessV0KeyHash(nodePtr->ownerAuthAddress));
-        targetHeight = ::ChainActive().Height() + 1;
+        targetHeight = chainHeight(*pwallet->chain().lock()) + 1;
```
Comment (ShengguangXiao) [inline src/masternodes/mn_rpc.cpp:315]: What's the purpose of this change? In line 309 the chain is locked already

**Comment by monstrobishi** [comment]
Can you confirm if this also allows `listtokens`, `listaccounts`, `gettoken` and `getaccount` to be called when the wallet is locked?

**Comment by monstrobishi** [comment]
I have updated the title here. Please provide a more descriptive title in the next PR :)

File: src/httpserver.cpp (hunk 1 above)
Comment (bvbfan) [reply src/httpserver.cpp:18]: It's used but not included, probably someone include it indirectly but now it doesn't, so it wont compile
https://github.com/DeFiCh/ain/pull/92/files/10eff033afbd58dd9c77646803b70a6c84d787a5#diff-63c8cb9c9dd61d50d59afd5c39914e1c259f8743030b637a7896a0746c851ef1R76

File: src/masternodes/mn_rpc.cpp (hunk 2 above)
Comment (bvbfan) [reply src/masternodes/mn_rpc.cpp]: You're right, revert.

**Comment by bvbfan** [comment]
> Can you confirm if this also allows `listtokens`, `listaccounts`, `gettoken` and `getaccount` to be called when the wallet is locked?

I can confirm.

**Comment by monstrobishi** [comment]
Looks good

File: src/masternodes/accounts.h [hunk 3]
```diff
@@ -14,7 +14,7 @@
 class CAccountsView : public virtual CStorageView
 {
 public:
-    void ForEachBalance(std::function<bool(CScript const & owner, CTokenAmount const & amount)> callback, BalanceKey start) const;
+    void ForEachBalance(std::function<bool(CScript const & owner, CTokenAmount const & amount)> callback, BalanceKey start = {}) const;
```
Comment (monstrobishi) [inline src/masternodes/accounts.h:17]: Will this affect consensus?

Comment (bvbfan) [reply src/masternodes/accounts.h:17]: No, AFAIK.
